filament/backend/src/vulkan/VulkanStagePool.cpp

```diff
@@ -141,27 +194,34 @@ void VulkanStagePool::gc() noexcept {
     }
     const uint64_t evictionTime = mCurrentFrame - TIME_BEFORE_EVICTION;
 
-    // Destroy buffers that have not been used for several frames.
-    decltype(mFreeStages) freeStages;
-    freeStages.swap(mFreeStages);
+    decltype(mStages) freeStages;
+    freeStages.swap(mStages);
+    uint8_t freeStageCount = 0;  // Assuming we'll never have > 255 free stages
     for (auto pair : freeStages) {
```

Comment:
If you want to move `pair.second` don't you want `auto&` here instead?  Otherwise you're moving a copy?

Reply:
I think there's another place where `destroyStage` is called and also has the same pattern.